Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseCollectionStateManager.cs

```diff
@@ -125,8 +146,33 @@ public TimeSpan UpdateState(string instrumentationKey)
                     // no samples to submit, do nothing
                     return this.DetermineBackOffs();
                 }
+                else
+                {
+                    // we have samples
+                    if (dataSamplesToSubmit.Any(sample => Interlocked.Read(ref sample.CollectionConfigurationAccumulator.ReferenceCount) != 0))
+                    {
+                        ////!!! better solution?
+                        // some samples are still being processed, wait a little to give them a chance to finish
+                        Thread.Sleep(this.coolDownTimeout);
+
+                        bool allCooledDown =
+                            dataSamplesToSubmit.All(sample => Interlocked.Read(ref sample.CollectionConfigurationAccumulator.ReferenceCount) == 0);
+
+                        QuickPulseEventSource.Log.TroubleshootingMessageEvent(
+                            string.Format(
```

Comment:
do not use `string.Format` in event source messages. Event source already allows to have parametrized messages